cylc/flow/scripts/remove.py

```diff
@@ -18,7 +18,27 @@
 
 """cylc remove [OPTIONS] ARGS
 
-Remove one or more task instances from a running workflow.
+Remove tasks from a running workflow and the workflow's history.
+
+By default, the specified task(s) will be removed from all flows, making it
+as if the task(s) never ran. The task(s) and their outputs will be left in the
+`None` flow to preserve a record that they ran, however.
+
+If you specify flow(s) to remove a task from (not all flows that it belongs to)
+then the task will still exist in the remaining flow(s), but it will not
+influence the evolution of the workflow in the specified flow(s).
+
+Removing a submitted or running task will also kill it (see "cylc kill").
+
+Examples:
+  # Remove a task that already ran.
+  # (Any downstream tasks that are already running or finished will be
+  # left alone. The task and its outputs will be left in the None flow)
```

Comment:
The bit in parentheses is quite important and should probably be in the main help text where it mentions downstream tasks.

However, this relates to my question on the scheduler module changes above. Why do we not kill downstream tasks that are already running, if they end up with no flow numbers? 

From a quick look back at the proposal it doesn't seem to explicitly address this for downstream tasks:

> If removing a task causes all of the prerequisites on a downstream task to be unset (i.e. if the downstream task was spawned as a result of outputs from this task alone) then the downstream task shall be removed from the pool

.., but the general kill statement would seem to apply equally well to downstream tasks too:

> When all flows are removed from active tasks, Cylc should first attempt to kill the job as it serves no further purpose and may cause resource contention issues if left running

Might be a question for @oliver-sanders?

Reply:
OK makes sense.  Can we amend the proposal to record this, to avoid having to think through it again in the future?

Reply:
Done.